ui/src/app/workflows/components/workflow-details/artifact-panel.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import {useEffect, useState} from 'react';
+import MonacoEditor from 'react-monaco-editor';
+import {Artifact, Workflow} from '../../../../models';
+import {artifactKey} from '../../../shared/artifacts';
+import ErrorBoundary from '../../../shared/components/error-boundary';
+import {ErrorNotice} from '../../../shared/components/error-notice';
+import {FirstTimeUserPanel} from '../../../shared/components/first-time-user-panel';
+import {GiveFeedbackLink} from '../../../shared/components/give-feedback-link';
+import {LinkButton} from '../../../shared/components/link-button';
+import {services} from '../../../shared/services';
+import requests from '../../../shared/services/requests';
+
+export const ArtifactPanel = ({
+    workflow,
+    artifact,
+    archived
+}: {
+    workflow: Workflow;
+    artifact: Artifact & {nodeId: string; artifactNameDiscriminator: string};
+    archived?: boolean;
+}) => {
+    const downloadUrl = services.workflows.getArtifactDownloadUrl(workflow, artifact.nodeId, artifact.name, archived, artifact.artifactNameDiscriminator === 'input');
+
+    const key = artifactKey(artifact)
+        .split('/')
```

Comment:
Is there a system agnostic way to do this?

Reply:
these are S3 keys, so always use `/` not `\\`?